### ISSUE TITLE
tar.zst doesn't seem supported

### ISSUE THREAD

**Issue by kivutar** [mholt/archiver#285]
## What version of the package or command are you using?

I did a fresh `go get github.com/mholt/archiver/cmd/arc`

## What are you trying to do?

Compress a file to tar.zst

## What steps did you take?

 1. `go get github.com/mholt/archiver/cmd/arc`
 2. arc compress foo tar.zst
 3. `format unrecognized by filename: tar.zst`

## What did you expect to happen, and what actually happened instead?

I expect to get a tar.zst archive. I got an error message instead.

**Comment by pratikbalar** [comment]
I am getting 
```log
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x57054e]

goroutine 1 [running]:
github.com/mholt/archiver/v3.(*TarZstd).wrapWriter(0xc00013deb8)
        /home/pratik/go/pkg/mod/github.com/mholt/archiver/v3@v3.5.1/tarzst.go:77 +0x6e
github.com/mholt/archiver/v3.(*TarZstd).Archive(0xc00013deb8, {0xc00013dee0, 0x1, 0x1}, {0x5c3acb, 0x19})
        /home/pratik/go/pkg/mod/github.com/mholt/archiver/v3@v3.5.1/tarzst.go:35 +0xa5
main.main()
        /home/pratik/workspace/pratikbalar/local-cache/main.go:29 +0x27d
exit status 2

```

**Comment by mholt** [comment]
@pratikbalar You need to initialize your TarZstd struct, look at the godoc for its embedded field.

**Comment by mholt** [comment]
This will be fixed by #302 (albeit the command has yet to be implemented, but the whole package has been rewritten so this won't be a problem).
